lib/delighted/resource.rb

```diff
@@ -11,6 +11,20 @@ def expandable_attributes
       def singleton_resource?
         !!@singleton_resource
       end
+
+      def identifier_string(id_hash)
+        raise ArgumentError, "must pass Hash" unless Hash === id_hash
+
+        id_key = id_hash.keys.detect { |k| !id_hash[k].to_s.empty? }
+        raise ArgumentError, "must pass an identifier name and value" unless id_key
+        id_value = id_hash[id_key]
+
+        if id_key.to_s == "id"
+          id_value
```

Comment:
Maybe `#to_s` this to ensure this method always returns a `String` and cover folks passing numeric IDs.